src/components/kepler-gl.js

```diff
@@ -333,67 +335,71 @@ function KeplerGlFactory(
       const theme = this.availableThemeSelector(this.props);
 
       return (
-        <ThemeProvider theme={theme}>
-          <GlobalStyle
-            width={width}
-            height={height}
-            className="kepler-gl"
-            id={`kepler-gl__${id}`}
-            ref={this.root}
-          >
-            <NotificationPanel {...notificationPanelFields} />
-            {!uiState.readOnly && <SidePanel {...sideFields} />}
-            <div className="maps" style={{display: 'flex'}}>
-              {mapContainers}
-            </div>
-            {isExporting && (
-              <PlotContainer
-                width={width}
-                height={height}
-                exportImageSetting={uiState.exportImage}
-                mapFields={mapFields}
-                addNotification={uiStateActions.addNotification}
-                startExportingImage={uiStateActions.startExportingImage}
-                setExportImageDataUri={uiStateActions.setExportImageDataUri}
-                setExportImageError={uiStateActions.setExportImageError}
+        <RootContext.Provider value={this.root}>
```

Comment:
why are we passing this context?
is to just support the portaled component?

Reply:
yes, so all the modal from Portabled will be mounted under kepler.gl component